contracts/ExitFormat.sol

```diff
@@ -23,14 +17,26 @@ library ExitFormat {
 
     // A SingleAssetExit specifies
     // * an asset address (0 implies the native asset of the chain: on mainnet, this is ETH)
-    // * custom metadata (optional field, can be zero bytes). This might specify how to transfer this particular asset (e.g. target an "ERC20.transfer"' method)
+    // * custom metadata (optional field, can be zero bytes).
+    //   the first byte of the metadata indicates how the metadata should be 
+    //   interpreted as per the ExitMetadataType enum. 
+    //   - empty metadata or metadata[0] == 0 => ERC20
+    //   - metadata[1] == 1 => ERC1155
```

Comment:
The `Allocation` struct includes an `allocationType` as well as `metadata`. This allows the `allocationType` to serve the purpose that you use the first byte for. Did you consider that approach?

I believe it would resolve the issue you pointed out [here](https://github.com/statechannels/exit-format/pull/69#discussion_r885159225), and it seems to be a bit easier to understand.

Reply:
I don't think this approach can work. If I understand correctly there still needs to be a way to unambiguously refer to assets for `guarantee` exit types to allow virtual channels